kernel/poll.c

```diff
@@ -213,12 +213,10 @@ static inline int register_events(struct k_poll_event *events,
 			set_event_ready(&events[ii], state);
 			poller->is_polling = false;
 		} else if (!just_check && poller->is_polling) {
-			int rc = register_event(&events[ii], poller);
-			if (rc == 0) {
-				events_registered += 1;
-			} else {
-				__ASSERT(false, "unexpected return code\n");
-			}
+
+			__ASSERT(register_event(&events[ii], poller) == 0, "unexpected return code\n");
```

Comment:
If assertions are disabled, register_event() is not called at all, see https://github.com/zephyrproject-rtos/zephyr/blob/master/include/sys/__assert.h#L111

Or am i missing something?